File: app/src/main/java/com/example/team_23_project/SearchBar.java
package com.example.team_23_project;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.SearchView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;


public class SearchBar extends AppCompatActivity {

    ListView listView;
    String[] universities = {"Newcastle University", "Northumbria University", "University of Bristol",
            "UCL", "Oxford", "Cambridge",
            "University of Manchester", "University of Liverpool", "University of Birmingham",
            "University of Aberdeen", "University of Bath", "University of Brighton", "University of" +
            " East Anglia", "University of Edinburgh", "University of Glasgow", "Imperial College London",
            "University of Leeds", "University of London", "University of Nottingham", "Swansea University",
            "University of Warwick", "University of York", "University of Westminster", "University of West London"};

    ArrayAdapter<String> arrayAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_searchbar);

        listView = findViewById(R.id.listView);

        arrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, universities);
        listView.setAdapter(arrayAdapter);


        listView.setOnItemClickListener((parent, view, position, id) -> { // <- Event handler for items in the list
            String selectedItem = (String) parent.getItemAtPosition(position); // <- Item the user
                                                                               // clicks on as a
                                                                               // String (e.g. will
                                                                               // output 'Oxford'
                                                                               // if user selects oxford)
                if (!selectedItem.equals("Newcastle University")){
                    Toast.makeText(SearchBar.this, "We're sorry, but we do not have support for this University yet",
                            Toast.LENGTH_SHORT).show();
                } else {
                    Intent intent = new Intent(SearchBar.this, SearchBarBuildings.class);
                    SearchBar.this.startActivity(intent);
                }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.nav_menu, menu);

        MenuItem menuItem = menu.findItem(R.id.nav_search);
        SearchView searchView = (SearchView) menuItem.getActionView();
        searchView.setIconified(false);
        searchView.setQueryHint("Search University");

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                arrayAdapter.getFilter().filter(newText);
                return false;
            }
        });
        return super.onCreateOptionsMenu(menu);
    }
}

File: app/src/main/java/com/example/team_23_project/SearchBar.kt
package com.example.team_23_project

import android.os.Bundle
import android.os.PersistableBundle
import android.view.Menu
import android.view.MenuItem
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class SearchBar : AppCompatActivity() {

    private lateinit var adapter: ArrayAdapter<*>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, resources.getStringArray(R.array.item_list))
        lv_listView.adapter = adapter
        lv_listView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            Toast.makeText(applicationContext, parent?.getItemAtPosition(position).toString(), Toast.LENGTH_SHORT).show()
        }
        lv_listView.emptyView = tv_emptyTextView

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.nav_menu, menu)

        val search : MenuItem? = menu?.findItem(R.id.nav_search)
        val searchView : SearchView = search?.actionView as SearchView
        searchView.queryHint = "Search Universities"

        searchView.setOnQueryTextListener(object: SearchView.OnQueryTextListener{
            override fun onQueryTextSubmit(p0: String?): Boolean {
                return false
            }

            override fun onQueryTextChange(p0: String?): Boolean {
                adapter.filter.filter(newText)
                return true
            }

        })
        return super.onCreateOptionsMenu(menu)
    }

}
